control-plane/subcommand/inject-connect/command.go

```diff
@@ -238,9 +251,21 @@ func (c *Command) Run(args []string) int {
 		return 1
 	}
 
+	// TODO (ishustava): eventually we will use go-netaddr library which doesn't use hclog,
+	// and so this additional logger will go away and we'll be able to use zap logger.
+	hclogger, err := common.Logger(c.flagLogLevel, c.flagLogJSON)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Unable to create logger: %s", err))
+		return 1
+	}
+	serverAddresses, err := common.GetResolvedServerAddresses(c.flagServerAddresses, c.providers, hclogger)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Unable to discover any Consul addresses from %q: %s", c.flagServerAddresses[0], err))
+		return 1
+	}
```

Comment:
This code is not unit-tested because the testing is involved and it will be switched out for the go-netaddr library before merging to `main`. I've added the cloud auto-join code here for consistency because currently `externalServers.hosts` value supports it.